### PR TITLE
Update Python and Poetry to fix commit errors

### PR DESCRIPTION
**Pull request by samfulcrum**
1. Updated Python from 3.6 / 3.7 ->  3.9.16
2. Updated Poetry 1.1 -> 1.1.15
3. Added instructions in README for mamba install in the case of the OSError shown below which was thrown when using conda. 

`OSError
Could not find a suitable TLS CA certificate bundle, invalid path: /Users/---/mambaforge/envs/pybedlite/lib/python3.9/site-packages/certifi/cacert.pem`

### CODE REVIEW

File: .github/workflows/pythonpackage.yml [hunk 1]
```diff
@@ -2,14 +2,14 @@ name: Python package
...
-  POETRY_VERSION: 1.1
+  POETRY_VERSION: 1.1.15
 
 jobs:
   testing:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        PYTHON_VERSION: ["3.6", "3.7"]
+        PYTHON_VERSION: ["3.9.16"]
```
Comment (nh13) [inline .github/workflows/pythonpackage.yml:12]: But then we aren't testing support for 3.6 or 3.7.  I know 3.6 is end of life already, and 3.7 in a few months, but if folks who rely on this library are on 3.6, then we really should be testing it!  And if we want to move on from 3.6, we need to do so [here](https://github.com/fulcrumgenomics/pybedlite/blob/a2660a452fe38794324996ebe3bee6fc4b665f91/pyproject.toml#L28) and make a big announcement.

**Review by nh13 — CHANGES_REQUESTED** [review]
We need to discuss internally with @tfenne about which python versions we need to support

File: .github/workflows/pythonpackage.yml (hunk 1 above)
Comment (samfulcrum) [reply .github/workflows/pythonpackage.yml]: @nils please see this commit https://github.com/fulcrumgenomics/pybedlite/pull/11/commits/5b6c30fd981c400e3162a6bc0eba7659d5ea0b3b specifying ubuntu-20.4 allows for testing of 3.6 and 3.7 :)

**Review by nh13 — APPROVED** [review]

**Comment by nh13** [comment]
@samfulcrum ready to merge!

**Pull request merged**